lib/dpi_dasm/dpi_dasm.cxx

```diff
@@ -109,10 +110,10 @@ extern "C" const char* get_insn_name(uint64_t insn) {
     initialize_disassembler();
   }
 
-  insn            = endian_swap(insn, gv_is_big_endian);
-  string insn_str = gp_disassembler->lookup(insn & 0xFFFFFFFF)->get_name();
-
-  return insn_str.data();
+  insn       = endian_swap(insn, gv_is_big_endian);
+  gs_ret_str = gp_disassembler->lookup(insn & 0xFFFFFFFF)->get_name();
```

Comment:
Any reason why we have to use a global variable here?

Reply:
Yes, when running in Questa and passing back a const char* from a stack-based string object the char* buffer was regularly corrupted.  The global ensures the c_str() buffer is persistent though the SystemVerilog side copying out the string buffer.